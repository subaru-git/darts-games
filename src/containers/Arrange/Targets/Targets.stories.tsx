/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Targets from './Targets';

export default {
  title: 'Components/Targets',
  component: Targets,
} as ComponentMeta<typeof Targets>;

const Template: ComponentStory<typeof Targets> = (args) => <Targets {...args} />;

export const Default = Template.bind({});
Default.args = {};
