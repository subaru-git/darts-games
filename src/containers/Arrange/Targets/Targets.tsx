import React, { FC } from 'react';
import { Grid, Text } from '@chakra-ui/react';

type TargetsProps = {
  count: number;
  targets: number[];
  isFinished?: boolean;
};

const Targets: FC<TargetsProps> = ({ count, targets, isFinished = false }) => {
  const rest = () => {
    const t = isFinished ? targets : targets.slice(0, -1);
    return new Array(count).fill(' - ').map((_, i) => t[i] ?? '-');
  };
  return (
    <Grid
      templateColumns={{
        base: `15px repeat(${count}, 27px) 15px`,
        md: `20px repeat(${count}, 38px) 20px`,
      }}
      gap={1}
      textAlign='center'
    >
      <Text fontWeight='bold' fontSize={{ base: 'sm', md: 'lg' }} color='gray.500'>
        {'['}
      </Text>
      {rest().map((t, i) => (
        <Text key={i} fontWeight='bold' fontSize={{ base: 'sm', md: 'lg' }} color='gray.500'>
          {t}
        </Text>
      ))}
      <Text fontWeight='bold' fontSize={{ base: 'sm', md: 'lg' }} color='gray.500'>
        {']'}
      </Text>
    </Grid>
  );
};

export default Targets;
