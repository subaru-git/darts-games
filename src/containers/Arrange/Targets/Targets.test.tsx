import { render } from '@testing-library/react';
import Targets from './Targets';

test('should rendering', () => {
  const { container } = render(<Targets count={8} targets={[]} isFinished={false} />);
  expect(container).toMatchSnapshot();
});
