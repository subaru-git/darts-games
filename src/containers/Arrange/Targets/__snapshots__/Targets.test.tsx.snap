// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-17ur22"
  >
    <p
      class="chakra-text css-18kwk26"
    >
      [
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      -
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      -
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      -
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      -
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      -
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      -
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      -
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      -
    </p>
    <p
      class="chakra-text css-18kwk26"
    >
      ]
    </p>
  </div>
</div>
`;
